### PR TITLE
GPII-2544 - Set the fqdn as NODENAME by default

### PR DESCRIPTION
**Pull request by amatas**
The Nodename must be the same along all the CouchDB cluster, by default the Nodename in Kubertenetes is 'noname' in all the containers. Thus the cluster is unable to be set properly.

With this code the default Nodename will be the fqdn of the container inside the Kubernetes cluster, thanks to this all the CouchDB containers will have an unique and consistent hostname, which will be a requirement to setup the final CouchDB cluster.

### CODE REVIEW

File: 2.1.0/docker-entrypoint.sh [hunk 1]
```diff
@@ -36,6 +36,9 @@ if [ "$1" = '/opt/couchdb/bin/couchdb' ]; then
 
 	if [ ! -z "$NODENAME" ] && ! grep "couchdb@" /opt/couchdb/etc/vm.args; then
 		echo "-name couchdb@$NODENAME" >> /opt/couchdb/etc/vm.args
+    else
```
Comment (mrtyler) [inline 2.1.0/docker-entrypoint.sh]: Indentation of this section looks wrong.

File: 2.1.0/docker-entrypoint.sh [hunk 2]
```diff
@@ -36,6 +36,9 @@ if [ "$1" = '/opt/couchdb/bin/couchdb' ]; then
...
+        NODENAME=$(hostname -f)
+        echo "-name couchdb@$NODENAME" >> /opt/couchdb/etc/vm.args
```
Comment (mrtyler) [inline 2.1.0/docker-entrypoint.sh]: Are you sure this logic is right? I'm not sure I understand the context but it looks like this stanza will write a duplicate `-name` line if `NODENAME` is non-empty but `couchdb@` was already in vm.args.

Maybe refactor to:
```
if ! grep "couchdb@" ...; then
  if [ -z "$NODENAME" ]; then
    NODENAME=$(hostname -f)
  fi
  echo "-name couchdb@$NODENAME" ...
fi

**Review by mrtyler — COMMENTED** [review]
1. Do you plan to get this change accepted back to upstream? Do you plan to run our own builds of the couchdb docker image until the change is merged? What is the plan? :)

2. Looks like you've made changes that will affect couchdb 2. The last time we discussed this, I thought we agreed to stick with couchdb 1 for the October release. Did something change? Am I missing something?

File: 2.1.0/docker-entrypoint.sh (hunk 1 above)
Comment (amatas) [reply 2.1.0/docker-entrypoint.sh]: image
Damn tabs vs spaces....

**Comment by amatas** [comment]
1. The FQDN domain name is needed to create the CouchDB cluster in Kubernetes. I'm not sure if this is a feature that upstream would be interested in. My plan is to connect this repository with our CI and rebuild the image everytime that CouchDB releases a new version.
2. The CouchDB 1 image doesn't need any changes. We can use the ones from the official library: https://hub.docker.com/_/couchdb/

File: 2.1.0/docker-entrypoint.sh [hunk 3]
```diff
@@ -15,63 +15,64 @@ set -e
...
+    chmod 775 /opt/couchdb/etc/*.d
 
-	if [ ! -z "$NODENAME" ] && ! grep "couchdb@" /opt/couchdb/etc/vm.args; then
-		echo "-name couchdb@$NODENAME" >> /opt/couchdb/etc/vm.args
-    else
+    if [ -z "$NODENAME" ]; then
         NODENAME=$(hostname -f)
+    fi
+
+    if [ ! -z "$NODENAME" ] && ! grep "couchdb@" /opt/couchdb/etc/vm.args; then
```
Comment (mrtyler) [inline 2.1.0/docker-entrypoint.sh]: I guess you can drop `[ ! -z "$NODENAME" ]` since we just set NODENAME :).

**Comment by mrtyler** [comment]
Thanks, your plan seems reasonable.

LGTM.

File: 2.1.0/docker-entrypoint.sh (hunk 3 above)
Comment (amatas) [reply 2.1.0/docker-entrypoint.sh]: Well, I thought on the case that the `hostname` command returns nothing.

Comment (mrtyler) [reply 2.1.0/docker-entrypoint.sh]: Ah, ok.

I think `-n` is more canonical than `[! -z]` :)

**Comment by mrtyler** [comment]
Thinking about this more, I think your whitespace changes are going to make it difficult to merge changes from upstream. If it were me, I'd back out the whitespace changes to prevent conflicts in the future. Up to you how to proceed.

File: 2.1.0/docker-entrypoint.sh (hunk 3 above)
Comment (mrtyler) [reply 2.1.0/docker-entrypoint.sh]: Please stop using `if ! -z` :)

Should this have the `if ! grep $COOKIE` logic to prevent against duplicate entries?

**Comment by amatas** [comment]
I've reverted the indentation using whitespaces in order to allow a possible merge with upstream.

**Comment by mrtyler** [comment]
Is this PR still relevant?

**Comment by amatas** [comment]
This PR is not still relevant.

**Pull request closed without merging**